docs-specs-generator/spec_file.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	internalAttributes = []string{"newrelic.integrationName", "newrelic.integrationVersion", "newrelic.source", "newrelic.agentVersion"}
+	ignoredAttributes  = []string{"agentName", "coreCount", "processorCount", "systemMemoryBytes"}
+)
+
+func generateSpecFile(c Config, metrics []Metric, filename string) *Specs {
+	sp := Specs{
+		SpecVersion:                  "2",
+		OwningTeam:                   "integrations",
+		IntegrationName:              c.IntegrationName,
+		HumanReadableIntegrationName: c.HumanReadableIntegrationName,
+		Entities:                     nil,
+	}
+	for _, m := range metrics {
+		entityType, ok := getMatchingEntity(c.Definitions, m.name, m.attributes)
+		if !ok {
+			fmt.Printf("metric not matching %s \n", m.name)
+			continue
+		}
+
+		e, ok := isEntityDefined(sp.Entities, entityType)
+		if !ok {
+			e = &Entity{
+				EntityType: entityType,
+			}
+			sp.Entities = append(sp.Entities, e)
+		}
+
+		if ok := isMetricDefined(e.Metrics, m.name); ok {
+			continue
+		}
+
+		mSpec := MetricSpec{
+			Name:              m.name,
+			Type:              string(m.metricType),
+			DefaultResolution: 15,
+			Unit:              "count",
+			Description:       m.description,
+			Dimensions:        nil,
+		}
+		for k, _ := range m.attributes {
+			mSpec.Dimensions = append(mSpec.Dimensions, Dimension{
+				Name: k,
+				Type: "string",
+			})
+		}
+
+		if mSpec.Type == string(metricType_HISTOGRAM) {
+			e.Metrics = append(e.Metrics, computeHistogramMetrics(mSpec)...)
+		} else {
+			e.Metrics = append(e.Metrics, &mSpec)
+		}
```

Comment:
It would be good if the `Metric` object could have a `asNRMetric()` method, so it could transform itself into an NR metric.